Data pipeline

nlubalo opened the pull request:
Review the new updated db_script.py.

nlubalo committed:
data-pipeline

nlubalo committed:
Delete cred.json

nlubalo committed:
Update .gitignore

nlubalo committed:
set pipenv

nlubalo committed:
Merge branch 'data-pipeline' of https://github.com/DalbergDataInsights/icohs-analysis into data-pipeline

nlubalo committed:
Update .gitignore

nlubalo committed:
Update .gitignore

nlubalo committed:
Update .gitignore

nlubalo committed:
Update .gitignore

nbigaud committed:
Updated structure of the json repository of pathes to data

nbigaud committed:
Updated structure of cleaning pipeline

nbigaud committed:
Adding autopep8 to pipfile

Adding autopep8 to pipfile

nbigaud committed:
Modify path to engine to fit dropbox setup

nbigaud committed:
Adding venv

nbigaud committed:
Updated path to data to reflect dropbox set up

nbigaud committed:
Cache change - unsure the exact purpose

nbigaud committed:
Update indicators.json

nbigaud committed:
Update preprocess.py

nbigaud committed:
preprocess.py : Fixing dtype warning at import

nbigaud committed:
Update preprocess.py

nbigaud committed:
Temporary path changes - for test phase

nbigaud committed:
Update preprocess.py

nbigaud committed:
Defining path for test datasets

nbigaud committed:
Streamlining code

nbigaud committed:
Update preprocess.py

Solved bug doubling certain numbers

nbigaud committed:
Preprocess: WIP update of the breakdown in four tables

nbigaud committed:
Update preprocess.py

nbigaud committed:
Update indicators.json

nbigaud committed:
Update preprocess.py

nbigaud committed:
Update .gitignore

nbigaud committed:
Updated pathes for testing purposes

nbigaud committed:
Update indicators.json

s

nbigaud committed:
Code updated for test on full data

nbigaud committed:
Update indicators.json

nbigaud committed:
Update preprocess.py

nbigaud committed:
Update engine.cpython-37.pyc

nbigaud committed:
Update preprocess.py

nbigaud committed:
Update preprocess.py

nlubalo committed:
update database script

nbigaud committed:
Preprocess : free up ressource use

nbigaud committed:
Update preprocess.py

nbigaud committed:
Merge branch 'data-pipeline' of https://github.com/DalbergDataInsights/icohs-analysis into data-pipeline

nbigaud committed:
Correcting for redundencies

nbigaud committed:
Note needed anymore

nbigaud committed:
Cleaning step code

nbigaud committed:
Processing step code added

nbigaud committed:
Added to be able to reference sub folder

nbigaud committed:
Orchastrator file added

nbigaud committed:
Added path to provisionnal intermediary csv

dsementsov commented on src/clean_and_process.py:
```diff
@@ -0,0 +1,65 @@
+from pipeline_func.process import process
+
+ENGINE = engine.get_engine()
+ENGINE_rename = engine.get_engine()
```
You don't need that variable

dsementsov commented on src/clean_and_process.py:
```diff
@@ -0,0 +1,65 @@
+ENGINE_rename = engine.get_engine()
+
+module_json = 'config/indicators.json'
+with open(module_json) as f:
+    for section in json.load(f):
+        for p in section['data']:
+            ENGINE[p['identifier']] = p['value']
+
```
This is a duplication of the get_engine file

dsementsov commented on src/clean_and_process.py:
```diff
@@ -0,0 +1,65 @@
+###########################
```
Can we have this file at the root of repo and rename it to app.py?

dsementsov commented on src/db_insert.py:
```diff
@@ -0,0 +1,67 @@
+import json 
+import sys
+
+def get_engine(json_file, sec):
+    with open(json_file) as f:
+        ENGINE = dict((p['identifier'], p['value']) for section in json.load(f) 
+                                                    for p in section[sec])
+        return ENGINE
```
This is a duplicate of the get_engine file

dsementsov commented on src/pipeline_func/clean.py:
```diff
@@ -0,0 +1,266 @@
+ENGINE = engine.get_engine()
+ENGINE_rename = engine.get_engine()
+
+module_json = 'config/indicators.json'
+with open(module_json) as f:
+    for section in json.load(f):
+        for p in section['data']:
+            ENGINE[p['identifier']] = p['value']
+
```
this is a duplicate of the get_engine

dsementsov commented on src/pipeline_func/clean.py:
```diff
@@ -0,0 +1,266 @@
+startTime = datetime.datetime.now()
+
+
+def make_note(statement):
+    print(statement, str(datetime.datetime.now() - startTime))
```
This method is duplicated. Move it to separate helpers.py file and import when needed.

dsementsov commented on src/pipeline_func/clean.py:
```diff
@@ -0,0 +1,266 @@
+
+# Get user-input data
+
+VAR_CORR_PATH = ENGINE['var_correspondance_data']
```
this variable is not needed. Just read from the key

dsementsov commented on src/pipeline_func/clean.py:
```diff
@@ -0,0 +1,266 @@
+# Selecting target indicators
+
+
+USECOLS = list(range(0, 10))
+DTYPES = {'Unnamed: 0': int, 'dataElement': str, 'period': str, 'orgUnit': str, 'categoryOptionCombo': str,
+          'attributeOptionCombo': str, 'value': object, 'storedBy': str, 'created': str, 'lastUpdated': str}
+
```
Constants should be at the top of the file

dsementsov commented on src/pipeline_func/clean.py:
```diff
@@ -0,0 +1,266 @@
+
+def get_data(path, instance):
+
+    data = pd.read_csv(path, usecols=USECOLS, dtype=DTYPES)
```
you can also set the index to be Unnamed: 0 right away

dsementsov commented on src/pipeline_func/clean.py:
```diff
@@ -0,0 +1,266 @@
+    if date.find("W") == -1:
+        x = pd.to_datetime(date, format='%Y%m')
+    else:
+        y = date.split("W")[0]
```
y, w = (date.split('W'))

dsementsov commented on src/pipeline_func/clean.py:
```diff
@@ -0,0 +1,266 @@
+
+def clean(new_dhis_path, old_dhis_path, new_dhis_report_path, old_dhis_report_path):
+
+    new_var_add_dict = get_variable_addition_dict(VAR_CORR, 'new')
+    old_var_add_dict = get_variable_addition_dict(VAR_CORR, 'old')
+
+    # Reporting data
+
+    new_dhis_report_df = get_reporting_data(new_dhis_report_path, "new")
+    make_note('new reporting data formatted for use')
+
+    old_dhis_report_df = get_reporting_data(old_dhis_report_path, "old")
+    make_note('old reporting data formatted for use')
+
```
Allocate data resources only when you need them!

If you want this to be less RAM efficient, I would even read-write the data in chunks.

You are doing:

read data 1 
read data 2 
read data 3

process data 1
process data 2 
process data 3

You need to be doing:

read data 1
process data 1
return data 1 or write data 1

read data 2 
process data 2
...

dsementsov commented on src/pipeline_func/clean.py:
```diff
@@ -0,0 +1,266 @@
+
+    # New instance
+
+    new_dhis_df = get_data(new_dhis_path, "new")
+
+    for x in list(new_var_add_dict.keys()):
+        new_dhis_df = compute_indicators(new_dhis_df, x, new_var_add_dict[x])
+    make_note('new data additional indicators created')
+
+    new_dhis_df = process_date(new_dhis_df)
+    make_note('new data transformed and cleaned')
+
+    # Old instance
+
+    old_dhis_df = get_data(old_dhis_path, "old")
+    for x in list(old_var_add_dict.keys()):
+        old_dhis_df = compute_indicators(old_dhis_df, x, old_var_add_dict[x])
+    make_note('old data additional indicators created')
+
+    old_dhis_df = process_date(old_dhis_df)
+    make_note('old data transformed and cleaned')
+
```
This is highly redundant and creates an overhead!

dsementsov commented on src/pipeline_func/clean.py:
```diff
@@ -0,0 +1,266 @@
+
+    # concatenate old and new instance
+
+    combined_df = pd.concat([old_dhis_df, new_dhis_df,
+                             old_dhis_report_df,
+                             new_dhis_report_df])
```
I am not sure we need reporting and data in the same df though

dsementsov commented on src/pipeline_func/clean.py:
```diff
@@ -0,0 +1,266 @@
+    make_note('district column added')
+
+    # Selecting only the facilities that are in both old and new
+    old_ids = set(old_dhis_report_df['orgUnit'].unique())
+    new_ids = set(new_dhis_report_df['orgUnit'].unique())
```
set and unique() does the same thing

dsementsov commented on src/pipeline_func/clean.py:
```diff
@@ -0,0 +1,266 @@
+    # Selecting only the facilities that are in both old and new
+    old_ids = set(old_dhis_report_df['orgUnit'].unique())
+    new_ids = set(new_dhis_report_df['orgUnit'].unique())
+    list_ids = list(new_ids.intersection(old_ids))
```
now what you could do is just 

set(old_dhis_report_df['orgUnit'].tolist() + new_dhis_report_df['orgUnit'].tolist())

dsementsov replied:
you also could have filtered the dfs before concatting them together

dsementsov commented on src/pipeline_func/process.py:
```diff
@@ -0,0 +1,280 @@
+from scipy import stats
+
+ENGINE = engine.get_engine()
+ENGINE_rename = engine.get_engine()
```
unused variable

dsementsov commented on src/pipeline_func/process.py:
```diff
@@ -0,0 +1,280 @@
+ENGINE = engine.get_engine()
+ENGINE_rename = engine.get_engine()
+
+module_json = 'config/indicators.json'
+with open(module_json) as f:
+    for section in json.load(f):
+        for p in section['data']:
+            ENGINE[p['identifier']] = p['value']
+
```
duplication of get_engine

dsementsov commented on src/pipeline_func/process.py:
```diff
@@ -0,0 +1,280 @@
+startTime = datetime.datetime.now()
+
+
+def make_note(statement):
+    print(statement, str(datetime.datetime.now() - startTime))
```
duplicate method

dsementsov commented on src/pipeline_func/process.py:
```diff
@@ -0,0 +1,280 @@
+# Get user-input data
+
+
+VAR_CORR_PATH = ENGINE['var_correspondance_data']
```
unnecessary declaration

dsementsov commented on src/pipeline_func/process.py:
```diff
@@ -0,0 +1,280 @@
+                         ['identifier'].unique())
+    data_df_noreport = data[~data['dataElement'].isin(
+        report_indics)].copy()
+    data_df_report = data[data['dataElement'].isin(report_indics)].copy()
```
you allocate memory 20 lines before you use it.

dsementsov committed:
altering git ignore

dsementsov committed:
deleting system files

dsementsov committed:
deleting __pycache

dsementsov committed:
refactor and code review

nbigaud committed:
Create variable_correspondance.csv

nbigaud committed:
Reflected changes in methods

nbigaud committed:
Further simplified teh function to correct error

nbigaud committed:
Update clean.py

Deleted references to district and change data parsing function

nbigaud committed:
process update

Deleted unnecessary output table and referecnes to districts

nbigaud committed:
Update variable_correspondance.csv

Addded description

nbigaud committed:
Update clean.py

Use all ids rather than select ids

nbigaud committed:
Update process.py

Reorders columns

nbigaud committed:
Update variable_correspondance.csv

Errors corrected in PCV/HPV/MR

nlubalo committed:
Delete db_insert.py

nlubalo committed:
add database adapter

nbigaud committed:
Update clean.py

bug fix l 150

nbigaud committed:
Update process.py

bug fix l150

dsementsov committed:
stash-empty-changes

dsementsov committed:
Merge branch 'db-adpater/data-pipeline' of https://github.com/DalbergDataInsights/icohs-analysis into data-pipeline

dsementsov committed:
merging db api to the pipeline

nbigaud committed:
Delete db_insert.py

Outdated file deleted

nbigaud committed:
Update app.py

Code commented out put back in

nbigaud committed:
Update app.py

Used the same stricture as the rest, to centralize all path references in the config

nbigaud committed:
Update indicators.json

Reviewed structure

nbigaud committed:
Update helpers.py

Reflected changes made to structure of the json file

nbigaud committed:
Update clean.py

Implmented selection of only facilities that are in the reporting data, still need to output missing IDs

nbigaud committed:
Update indicators.json

reformat according to Nancy's directions

nbigaud committed:
Update helpers.py

Coorected bug in the way the list was referenced

nlubalo committed:
fix json helper function

nbigaud committed:
Clean.py : started to udpate the code to run on all files found in the 'raw' folder. Still WIP, but I need to restart my computer

nbigaud committed:
App.py : update to account for new structure of the directory and functions

Using all data in teh 'raw' folder as input. WIP, not yet tested.

nbigaud committed:
indicators.json : added path to the raw data

nbigaud committed:
App.py : small correction to previus commit

nbigaud committed:
indicators.json : small correction to previous commit

nbigaud committed:
Adapter.py :  automatically linted by my VS code, no substancial changes

nbigaud committed:
Clean.py : WIP updated to work on a file by file basis and to include the db insertion step and also move the csv files

WIP, should work but not yet tested, will definitely throw errors. Need input from Nancy on the insert_clean_data function (line 267)

nbigaud committed:
App.py : decoupled lookuptable from the cleaning process (now done month by month) and solved dotenv import bug

nbigaud committed:
Clean.py : solved dotenv import issue and added facility lookup mapping step

nlubalo committed:
clean adapter

nlubalo committed:
Merge branch 'data-pipeline' of https://github.com/DalbergDataInsights/icohs-analysis into data-pipeline

nbigaud committed:
Process.py : minor type correction

nbigaud committed:
Adapter.py : no major changes

nbigaud committed:
Clean.py : made small changes to read the right data, tried to switch the writing function but is still not working

nbigaud committed:
Helper : automatically linted as I saved changes, no substancial changes

nbigaud committed:
App : updated the writing of the clean data to database, the overall structure, and added writing of the final table to the database

nbigaud committed:
Adapter: reviewed small points

nbigaud committed:
Clean : added assertition on year and month of data, restructured the sequence of cleaning and writing to a temporary csv

nbigaud committed:
Process: reflected changes in the SQL table structure

nbigaud committed:
Adapter : create new functions to write and delete lookup tables without breaking links with other databases

nbigaud committed:
Indictaors : updated path references

nbigaud committed:
Clean : corrected a small issues for reporting data (value>1 due to grouby of weekly data)

nbigaud committed:
App : reflected previous commits in the function called

nbigaud committed:
App : minor change

nbigaud committed:
Adapted : simplified and updated the read_table function

nbigaud committed:
Process : small correction made in pivoting outlier data to avoid creating empty rows

nbigaud committed:
App : restore code to production mode

nbigaud committed:
Create input_vs_output_data_check.ipynb

nbigaud committed:
Create test.py

nbigaud committed:
Create test_text.txt

nbigaud committed:
Renaiming file

nbigaud committed:
Create Untitled.ipynb

nbigaud committed:
Deleted rich text

nbigaud committed:
App : adding  the population data to the pipeline

nbigaud committed:
Indicators : adding path to population data

nbigaud committed:
Correcting a bug in lookup setup and creating an updte function for the pop data

nbigaud committed:
Clean : added op data cleaning function

nbigaud committed:
changing matching files structure to reflect addition of breakdown matching

nbigaud committed:
Added tests

nbigaud committed:
example breakdown added

nbigaud committed:
file to create facility lookup

nbigaud committed:
test file

nbigaud committed:
Testing and set up of breakdown tables

nbigaud committed:
support tool for manual creation of breakdown tables

nbigaud committed:
renaming config files

nbigaud committed:
added path to new config file

nbigaud committed:
First draft of a database structure for config file concatenation

nbigaud committed:
Added breakdown feature

nbigaud committed:
testing

nbigaud committed:
Added target

nbigaud committed:
Fixed bugs from new column names with suffixes using a helper function

nbigaud committed:
draft of helper function

nbigaud committed:
Added helper function

nbigaud committed:
Added a #TODO for what might be a bug when adding indicators

nbigaud committed:
dated the use of renaming dict to the new columns with suffix

nbigaud committed:
Adapted column references to the new columns with suffixes

nbigaud committed:
temporary tool to transform datasets before publication on Heroku DB

nbigaud committed:
SQL query to use (for now, still quite manual, would require automatic set up)

nbigaud committed:
Adapted for new populatiom database model

nbigaud committed:
draft of pop function

nbigaud committed:
Added path to pop percentages

nbigaud committed:
Updated query to new population data model

nbigaud committed:
Added new pop function

nbigaud committed:
Corrected typo

nbigaud committed:
Corrected typo

nbigaud committed:
added tester to gitignore

nbigaud committed:
Mi-changes for update of output

nbigaud committed:
Mid changes for updated output format

nbigaud committed:
Merge branch 'feature/update-data-output-format' into feature/breakdown-data

nbigaud committed:
Merge pull request #7 from DalbergDataInsights/feature/breakdown-data

Feature/breakdown data

nbigaud committed:
Uncommented code streamlined ordering of columns, and added new reporting category

nbigaud committed:
Added gender indics to config file, check if it runs

nbigaud committed:
Added changes to vit a

nbigaud committed:
Merge pull request #8 from DalbergDataInsights/feature/update-data-output-format

Feature/update data output format

nbigaud committed:
Merge pull request #9 from DalbergDataInsights/feature/first-gender-indics

Feature/first gender indics

nbigaud committed:
Corrected two bugs

nbigaud committed:
Correcting small bugs

nbigaud committed:
Check-point : working with a few minor bugs, requires error checking

nbigaud committed:
Checkpoint

nbigaud committed:
Added a tetsing folder to gitignore

nbigaud committed:
Working pipeline indic module, used to create indicators

nbigaud committed:
Added a feature passing on the config file to csv in output

nbigaud committed:
changed weight marker from " -- " to "__" due to SQL errors

nbigaud committed:
Small changes

nbigaud committed:
New indics added (nutrition ratios)

nbigaud committed:
Adjusments to fit nutrition team reco

nbigaud committed:
Correction to NUT indics

nbigaud committed:
Simplified output for memory issues

nbigaud committed:
Finalized changes

nbigaud committed:
uncommented app.py

nbigaud committed:
Merge pull request #10 from DalbergDataInsights/feature/add-calculated-indicators

Feature/add calculated indicators

nbigaud committed:
Reviewed indic files for new indicators

nbigaud committed:
small fix to pass on config

nbigaud committed:
switcched sum to nansum

nbigaud committed:
Merge branch 'data-pipeline' of https://github.com/DalbergDataInsights/icohs-analysis into feature/new-indics-who-gender

nbigaud committed:
Changes in names

nbigaud committed:
shortened coulm names to avoid issues in PostgresSQL DB down the line

nbigaud committed:
typo

nbigaud committed:
added TB gender breakdown

nbigaud commented:
Outdated I think
